data/campaigns/Dead_Water/scenarios/09_The_Mage.cfg

```diff
@@ -365,40 +434,203 @@
+            x,y=8,5
+        [/filter]
+        [unstore_unit]
+            variable=Caladon_stored
+            find_vacant=yes
+        [/unstore_unit]
+        [message]
+            speaker=Caladon
+            message= _ "What kind of abomination is THAT!?!"
+        [/message]
+        [animate_unit]
+            [filter]
+                id=Caladon
+            [/filter]
+            flag = attack
+            [primary_attack]
+                range=melee
+            [/primary_attack]
+            [facing]
+                x,y=$x1,$y1
+            [/facing]
+            hits=yes
+        [/animate_unit]
+        [move_unit]
+            x,y=$x1,$y1
+            to_x=8
+            to_y=6
+        [/move_unit]
+        [move_unit]
+            id=Caladon
+            to_x=8
+            to_y=5
+        [/move_unit]
+        [delay]
+            time=400
+            accelerate=yes
+        [/delay]
+        [store_unit]
+            [filter]
+                id=Caladon
+            [/filter]
+            variable=Caladon_stored
+            kill=yes
+        [/store_unit]
+        [message]
+            speaker=Cylanna
+            message= _ "It appears that Caladon does not want to deal with the undead." + " " + _ "He might be more welcoming of one of our people."
+        [/message]
+        [message]
+            speaker=Kai Krellis
+            message= _ "Let’s go, men!"
+        [/message]
+        [label]
+            x,y=8,5
+            text= _ "Caladon’s shack"
+        [/label]
+        [set_variable]
+            name=Caladon_discovered
+            value=yes
+        [/set_variable]
+        [show_objectives]
+        [/show_objectives]
+    [/event]
+
+    [event]
+        name=moveto
+        [filter]
+            side=1
             x=8
             y=5
-            side=1
-            id=Caladon
-            name= _ "Caladon"
-            unrenamable=yes
-            [modifications]
-                {TRAIT_LOYAL_HERO}
-                {TRAIT_QUICK}
-            [/modifications]
-            profile=portraits/caladon.webp
-        [/unit]
+            [not]
+                race=bats
+            [/not]
+            [not]
+                race=undead
+            [/not]
+        [/filter]
 
+        [unstore_unit]
+            variable=Caladon_stored
+            find_vacant=yes
+        [/unstore_unit]
         [if]
             [have_unit]
                 x,y=$x1,$y1
                 id=Keshan,Inky
             [/have_unit]
             [then]
-                [message]
-                    speaker=Keshan
-                    message= _ "There is somebody here."
-                [/message]
+                [if]
+                    [have_unit]
+                        x,y=$x1,$y1
+                        id=Keshan
+                    [/have_unit]
+                    [message]
+                        speaker=unit
+                        message= _ "There is somebody here."
+                    [/message]
```

Comment:
Surely this `[message]` should be in a `[then]` or does this actually work?

Reply:
@soliton- Can you please reopen and rebase this pull? I fixed that missing ```[then]```. Thank you

Reply:
I think since this has already been merged any further changes need to be in a new PR.